pallets/gear/src/tests.rs

```diff
@@ -12128,3 +12128,91 @@ fn calculate_gas_fails_when_calculation_limit_exceeded() {
         );
     });
 }
+
+#[test]
+fn reservation_manager() {
+    use demo_reservation_manager::{Action, WASM_BINARY};
+
+    init_logger();
+    new_test_ext().execute_with(|| {
+        let pid = Gear::upload_program(
+            RuntimeOrigin::signed(USER_1),
+            WASM_BINARY.to_vec(),
+            DEFAULT_SALT.to_vec(),
+            vec![],
+            BlockGasLimitOf::<Test>::get(),
+            0,
+        )
+        .map(|_| get_last_program_id())
+        .expect("Program uploading failed");
+
+        run_to_next_block(None);
+
+        let assert_for_expected_message = move |action, expected: Option<_>| {
+            System::reset_events();
+
+            assert_ok!(Gear::send_message(
+                RuntimeOrigin::signed(USER_1),
+                pid,
+                action,
+                BlockGasLimitOf::<Test>::get(),
+                0,
+            ));
+
+            run_to_next_block(None);
+
+            let expected_message_received = System::events().into_iter().any(|e| match e.event {
+                MockRuntimeEvent::Gear(Event::UserMessageSent { message, .. }) => {
+                    if message.destination().into_origin() == USER_1.into_origin() {
+                        expected.is_none() || Some(message.payload().to_vec()) == expected
+                    } else {
+                        false
+                    }
+                }
+                _ => false,
+            });
+
+            assert!(expected_message_received);
+
+            System::reset_events();
+        };
+
+        let assert_failed = move |action| {
```

Comment:
messing up naming, assert_failed already exists in the scope

Reply:
I'd go with inroducing some `fn scenario()`, see module entries for such "pattern"